test/bootloader_finish_client_test.rb

```diff
@@ -118,5 +119,24 @@
         subject.write
       end
     end
+
+    context "when there is no device to install grub" do
```

Comment:
well, if we allows to install grub without stage1, does it have warning in such case? as usually it does not work, unless user do own changes.